main.go

```diff
@@ -0,0 +1,55 @@
+// grep | (c) 2020 NETWAYS GmbH | GPLv2+
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+)
+
+func main() {
+	patternParameter := flag.String("e", "", "use PATTERN for matching.")
+	flag.Parse()
+
+	if *patternParameter != "" {
+		re, regErr := regexp.Compile(*patternParameter)
+		if regErr != nil {
+			fmt.Fprintln(os.Stderr, regErr)
+			os.Exit(1)
+		}
+
+		buf := bufio.NewReader(os.Stdin)
+
+		for {
+			data, dataErr := buf.ReadBytes('\n')
+			if dataErr != nil && dataErr != io.EOF {
+				fmt.Fprintln(os.Stderr, dataErr)
+				os.Exit(1)
+			}
+
+			if len(data) == 0 {
+				break
+			}
+
+			if !bytes.Contains(data, []byte{'\n'}) {
+				data = append(data, '\n')
+			}
+
+			if re.Match(data) {
+				_, _ = os.Stdout.Write(data)
+			}
+
+			if dataErr == io.EOF {
+				break
+			}
+		}
+	} else {
```

Comment:
Please put this out of the if (w/o changing the program's behavior) as in the other programs.